### ISSUE TITLE
JITaaS remote AOT mode doesn't handle delayed AOT load properly

### ISSUE THREAD

**Issue by harryyu1994** [eclipse/openj9#5131]
After an AOT compilation finishes, a lot of times AOT relocations are intentionally failed to delay the loading of AOT code. Current JITaaS remote AOT mode does not consider this case.

**Comment by harryyu1994** [comment]
Currently looking at this issue.

The following code snippet needs to be incorporated into JITaaS remote AOT
```
// AOT compilations can fail on purpose because we want to load
// the AOT body later on. This case is signalled by having a metaData != 0
// but a startPC == entry->_oldStartPC == 0
if (metaData && !entry->_oldStartPC && !startPC && _compInfo.getPersistentInfo()->getJITaaSMode() != SERVER_MODE)
   {
   TR_ASSERT(_vm->isAOT_DEPRECATED_DO_NOT_USE(), "compilationEnd() can fail only for relocating AOT compilations\n");
   TR_ASSERT(!entry->_oldStartPC, "We expect compilationEnd() to fail only for AOT compilations which are first time compilations\n");

   // Replenish the counts of the method
   // We are holding the compilation monitor at this point
   //
   J9ROMMethod *romMethod = J9_ROM_METHOD_FROM_RAM_METHOD(method);
   if (!(romMethod->modifiers & J9AccNative))// Never change the extra field of a native method
      {
      int32_t methodVMExtra = TR::CompilationInfo::getJ9MethodVMExtra(method);
      if (methodVMExtra == 1 || methodVMExtra == J9_JIT_QUEUED_FOR_COMPILATION)
         {
         // We want to use high counts unless the user specified counts on the command line
         // or he used useLowerMethodCounts (or Xquickstart)
         int32_t count;
         if (TR::Options::getCountsAreProvidedByUser() || TR::Options::startupTimeMatters() == TR_yes)
            count = getCount(romMethod, _compiler->getOptions(), _compiler->getOptions());
         else
            count = J9ROMMETHOD_HAS_BACKWARDS_BRANCHES(romMethod) ? TR_DEFAULT_INITIAL_BCOUNT : TR_DEFAULT_INITIAL_COUNT;

         TR::CompilationInfo::setInvocationCount(method, count);
         if (TR::Options::getCmdLineOptions()->getVerboseOption(TR_VerbosePerformance))
            {
            // compiler must exist because startPC != NULL
            TR_VerboseLog::writeLineLocked(TR_Vlog_PERF, "Reencoding count=%d for %s j9m=%p ", count, _compiler->signature(), method);
            }
         }
      else
         {
         if (_compInfo.getPersistentInfo()->getJITaaSMode() != SERVER_MODE)
            TR_ASSERT(false, "Unexpected value for method->extra = %p (method=%p)\n", TR::CompilationInfo::getJ9MethodExtra(method), method);
         }
      }
   }
```

**Comment by harryyu1994** [comment]
Fixed by #5143
